src/compute-types/src/dataflows.rs

```diff
@@ -726,106 +726,82 @@ impl Arbitrary for DataflowDescription<OptimizedMirRelationExpr, (), mz_repr::Ti
-                sink_descs.into_iter(),
-            ),
-            as_of: if as_of_some {
-                Some(Antichain::from(as_of))
-            } else {
-                None
-            },
-            until: Antichain::new(),
-            initial_storage_as_of: if initial_storage_as_of_some {
-                Some(Antichain::from(initial_as_of))
-            } else {
-                None
-            },
-            refresh_schedule: if refresh_schedule_some {
-                Some(refresh_schedule)
-            } else {
-                None
+fn any_dataflow_description<P, S, T>(
+    any_source_import: impl Strategy<Value = (GlobalId, (SourceInstanceDesc<S>, bool))>,
+) -> impl Strategy<Value = DataflowDescription<P, S, T>>
+where
+    P: Arbitrary,
+    S: 'static + Arbitrary,
+    T: Arbitrary + timely::PartialOrder,
+    ComputeSinkDesc<S, T>: Arbitrary,
+{
+    (
+        proptest::collection::vec(any_source_import, 1..3),
+        proptest::collection::vec(any_dataflow_index_import(), 1..3),
+        proptest::collection::vec(any::<BuildDesc<P>>(), 1..3),
+        proptest::collection::vec(any_dataflow_index_export(), 1..3),
+        proptest::collection::vec(any::<(GlobalId, ComputeSinkDesc<S, T>)>(), 1..3),
+        any::<bool>(),
+        proptest::collection::vec(any::<T>(), 1..5),
+        any::<bool>(),
+        proptest::collection::vec(any::<T>(), 1..5),
+        any::<bool>(),
+        any::<RefreshSchedule>(),
+        any::<Option<TimeDependence>>(),
+    )
+        .prop_map(
+            |(
+                source_imports,
+                index_imports,
+                objects_to_build,
+                index_exports,
+                sink_descs,
+                as_of_some,
+                as_of,
+                initial_storage_as_of_some,
+                initial_as_of,
+                refresh_schedule_some,
+                refresh_schedule,
+                time_dependence,
+            )| DataflowDescription {
+                source_imports: BTreeMap::from_iter(source_imports.into_iter()),
+                index_imports: BTreeMap::from_iter(index_imports.into_iter()),
+                objects_to_build,
+                index_exports: BTreeMap::from_iter(index_exports.into_iter()),
+                sink_exports: BTreeMap::from_iter(sink_descs.into_iter()),
+                as_of: if as_of_some {
+                    Some(Antichain::from(as_of))
+                } else {
+                    None
+                },
+                until: Antichain::new(),
+                initial_storage_as_of: if initial_storage_as_of_some {
+                    Some(Antichain::from(initial_as_of))
+                } else {
+                    None
+                },
+                refresh_schedule: if refresh_schedule_some {
+                    Some(refresh_schedule)
+                } else {
+                    None
+                },
+                debug_name: "debug_name".to_string(),
```

Comment:
I tried for awhile to try and generate a random string. I tried both `any::<String>()` and `proptest::string::string_regex(".*").unwrap(),`. Both approaches gave the following very helpful error:

```
/home/joe/.cargo/bin/cargo test --color=always --no-run --lib expr_cache::tests --profile test --manifest-path /home/joe/projects/materialize/src/catalog/Cargo.toml
   Compiling mz-compute-types v0.0.0 (/home/joe/projects/materialize/src/compute-types)
error[E0599]: the method `prop_map` exists for tuple `(VecStrategy<...>, ..., ..., ..., ..., ..., ..., ..., ..., ..., ..., ..., ...)`, but its trait bounds were not satisfied
   --> src/compute-types/src/dataflows.rs:771:10
    |
756 | /     (
757 | |         proptest::collection::vec(any_source_import, 1..3),
758 | |         proptest::collection::vec(any_dataflow_index_import(), 1..3),
759 | |         proptest::collection::vec(any::<BuildDesc<P>>(), 1..3),
...   |
770 | |     )
771 | |         .prop_map(
    | |         -^^^^^^^^ method cannot be called due to unsatisfied trait bounds
    | |_________|
    |
    |
    = note: the full type name has been written to '/home/mz-target/debug/deps/mz_compute_types-24500694e58e1d95.long-type-1348310666828065199.txt'
    = note: consider using `--verbose` to print the full type name to the console
    = note: the following trait bounds were not satisfied:
            `(VecStrategy<impl Strategy<Value = (GlobalId, (SourceInstanceDesc<S>, bool))>>, VecStrategy<impl _IMPL_ARBITRARY_FOR_IndexDesc::_proptest::strategy::Strategy<Value = (mz_repr::GlobalId, IndexImport)>>, VecStrategy<_IMPL_ARBITRARY_FOR_IndexDesc::_proptest::strategy::Map<(_IMPL_ARBITRARY_FOR_IndexDesc::_proptest::strategy::Union<_IMPL_ARBITRARY_FOR_IndexDesc::_proptest::strategy::BoxedStrategy<mz_repr::GlobalId>>, <P as _IMPL_ARBITRARY_FOR_IndexDesc::_proptest::arbitrary::Arbitrary>::Strategy), fn((mz_repr::GlobalId, P)) -> BuildDesc<P>>>, VecStrategy<impl _IMPL_ARBITRARY_FOR_IndexDesc::_proptest::strategy::Strategy<Value = (mz_repr::GlobalId, (IndexDesc, RelationType))>>, VecStrategy<(_IMPL_ARBITRARY_FOR_IndexDesc::_proptest::strategy::Union<_IMPL_ARBITRARY_FOR_IndexDesc::_proptest::strategy::BoxedStrategy<mz_repr::GlobalId>>, <ComputeSinkDesc<S, T> as _IMPL_ARBITRARY_FOR_IndexDesc::_proptest::arbitrary::Arbitrary>::Strategy)>, _IMPL_ARBITRARY_FOR_IndexDesc::_proptest::bool::Any, VecStrategy<<T as _IMPL_ARBITRARY_FOR_IndexDesc::_proptest::arbitrary::Arbitrary>::Strategy>, RegexGeneratorStrategy<std::string::String>, _IMPL_ARBITRARY_FOR_IndexDesc::_proptest::bool::Any, VecStrategy<<T as _IMPL_ARBITRARY_FOR_IndexDesc::_proptest::arbitrary::Arbitrary>::Strategy>, _IMPL_ARBITRARY_FOR_IndexDesc::_proptest::bool::Any, _IMPL_ARBITRARY_FOR_IndexDesc::_proptest::strategy::BoxedStrategy<RefreshSchedule>, OptionStrategy<_IMPL_ARBITRARY_FOR_IndexDesc::_proptest::strategy::BoxedStrategy<TimeDependence>>): _IMPL_ARBITRARY_FOR_IndexDesc::_proptest::strategy::Strategy`
            which is required by `&(VecStrategy<impl Strategy<Value = (GlobalId, (SourceInstanceDesc<S>, bool))>>, VecStrategy<impl _IMPL_ARBITRARY_FOR_IndexDesc::_proptest::strategy::Strategy<Value = (mz_repr::GlobalId, IndexImport)>>, VecStrategy<_IMPL_ARBITRARY_FOR_IndexDesc::_proptest::strategy::Map<(_IMPL_ARBITRARY_FOR_IndexDesc::_proptest::strategy::Union<_IMPL_ARBITRARY_FOR_IndexDesc::_proptest::strategy::BoxedStrategy<mz_repr::GlobalId>>, <P as _IMPL_ARBITRARY_FOR_IndexDesc::_proptest::arbitrary::Arbitrary>::Strategy), fn((mz_repr::GlobalId, P)) -> BuildDesc<P>>>, VecStrategy<impl _IMPL_ARBITRARY_FOR_IndexDesc::_proptest::strategy::Strategy<Value = (mz_repr::GlobalId, (IndexDesc, RelationType))>>, VecStrategy<(_IMPL_ARBITRARY_FOR_IndexDesc::_proptest::strategy::Union<_IMPL_ARBITRARY_FOR_IndexDesc::_proptest::strategy::BoxedStrategy<mz_repr::GlobalId>>, <ComputeSinkDesc<S, T> as _IMPL_ARBITRARY_FOR_IndexDesc::_proptest::arbitrary::Arbitrary>::Strategy)>, _IMPL_ARBITRARY_FOR_IndexDesc::_proptest::bool::Any, VecStrategy<<T as _IMPL_ARBITRARY_FOR_IndexDesc::_proptest::arbitrary::Arbitrary>::Strategy>, RegexGeneratorStrategy<std::string::String>, _IMPL_ARBITRARY_FOR_IndexDesc::_proptest::bool::Any, VecStrategy<<T as _IMPL_ARBITRARY_FOR_IndexDesc::_proptest::arbitrary::Arbitrary>::Strategy>, _IMPL_ARBITRARY_FOR_IndexDesc::_proptest::bool::Any, _IMPL_ARBITRARY_FOR_IndexDesc::_proptest::strategy::BoxedStrategy<RefreshSchedule>, OptionStrategy<_IMPL_ARBITRARY_FOR_IndexDesc::_proptest::strategy::BoxedStrategy<TimeDependence>>): _IMPL_ARBITRARY_FOR_IndexDesc::_proptest::strategy::Strategy`
            `(VecStrategy<impl Strategy<Value = (GlobalId, (SourceInstanceDesc<S>, bool))>>, VecStrategy<impl _IMPL_ARBITRARY_FOR_IndexDesc::_proptest::strategy::Strategy<Value = (mz_repr::GlobalId, IndexImport)>>, VecStrategy<_IMPL_ARBITRARY_FOR_IndexDesc::_proptest::strategy::Map<(_IMPL_ARBITRARY_FOR_IndexDesc::_proptest::strategy::Union<_IMPL_ARBITRARY_FOR_IndexDesc::_proptest::strategy::BoxedStrategy<mz_repr::GlobalId>>, <P as _IMPL_ARBITRARY_FOR_IndexDesc::_proptest::arbitrary::Arbitrary>::Strategy), fn((mz_repr::GlobalId, P)) -> BuildDesc<P>>>, VecStrategy<impl _IMPL_ARBITRARY_FOR_IndexDesc::_proptest::strategy::Strategy<Value = (mz_repr::GlobalId, (IndexDesc, RelationType))>>, VecStrategy<(_IMPL_ARBITRARY_FOR_IndexDesc::_proptest::strategy::Union<_IMPL_ARBITRARY_FOR_IndexDesc::_proptest::strategy::BoxedStrategy<mz_repr::GlobalId>>, <ComputeSinkDesc<S, T> as _IMPL_ARBITRARY_FOR_IndexDesc::_proptest::arbitrary::Arbitrary>::Strategy)>, _IMPL_ARBITRARY_FOR_IndexDesc::_proptest::bool::Any, VecStrategy<<T as _IMPL_ARBITRARY_FOR_IndexDesc::_proptest::arbitrary::Arbitrary>::Strategy>, RegexGeneratorStrategy<std::string::String>, _IMPL_ARBITRARY_FOR_IndexDesc::_proptest::bool::Any, VecStrategy<<T as _IMPL_ARBITRARY_FOR_IndexDesc::_proptest::arbitrary::Arbitrary>::Strategy>, _IMPL_ARBITRARY_FOR_IndexDesc::_proptest::bool::Any, _IMPL_ARBITRARY_FOR_IndexDesc::_proptest::strategy::BoxedStrategy<RefreshSchedule>, OptionStrategy<_IMPL_ARBITRARY_FOR_IndexDesc::_proptest::strategy::BoxedStrategy<TimeDependence>>): _IMPL_ARBITRARY_FOR_IndexDesc::_proptest::strategy::Strategy`
            which is required by `&mut (VecStrategy<impl Strategy<Value = (GlobalId, (SourceInstanceDesc<S>, bool))>>, VecStrategy<impl _IMPL_ARBITRARY_FOR_IndexDesc::_proptest::strategy::Strategy<Value = (mz_repr::GlobalId, IndexImport)>>, VecStrategy<_IMPL_ARBITRARY_FOR_IndexDesc::_proptest::strategy::Map<(_IMPL_ARBITRARY_FOR_IndexDesc::_proptest::strategy::Union<_IMPL_ARBITRARY_FOR_IndexDesc::_proptest::strategy::BoxedStrategy<mz_repr::GlobalId>>, <P as _IMPL_ARBITRARY_FOR_IndexDesc::_proptest::arbitrary::Arbitrary>::Strategy), fn((mz_repr::GlobalId, P)) -> BuildDesc<P>>>, VecStrategy<impl _IMPL_ARBITRARY_FOR_IndexDesc::_proptest::strategy::Strategy<Value = (mz_repr::GlobalId, (IndexDesc, RelationType))>>, VecStrategy<(_IMPL_ARBITRARY_FOR_IndexDesc::_proptest::strategy::Union<_IMPL_ARBITRARY_FOR_IndexDesc::_proptest::strategy::BoxedStrategy<mz_repr::GlobalId>>, <ComputeSinkDesc<S, T> as _IMPL_ARBITRARY_FOR_IndexDesc::_proptest::arbitrary::Arbitrary>::Strategy)>, _IMPL_ARBITRARY_FOR_IndexDesc::_proptest::bool::Any, VecStrategy<<T as _IMPL_ARBITRARY_FOR_IndexDesc::_proptest::arbitrary::Arbitrary>::Strategy>, RegexGeneratorStrategy<std::string::String>, _IMPL_ARBITRARY_FOR_IndexDesc::_proptest::bool::Any, VecStrategy<<T as _IMPL_ARBITRARY_FOR_IndexDesc::_proptest::arbitrary::Arbitrary>::Strategy>, _IMPL_ARBITRARY_FOR_IndexDesc::_proptest::bool::Any, _IMPL_ARBITRARY_FOR_IndexDesc::_proptest::strategy::BoxedStrategy<RefreshSchedule>, OptionStrategy<_IMPL_ARBITRARY_FOR_IndexDesc::_proptest::strategy::BoxedStrategy<TimeDependence>>): _IMPL_ARBITRARY_FOR_IndexDesc::_proptest::strategy::Strategy`

For more information about this error, try `rustc --explain E0599`.
error: could not compile `mz-compute-types` (lib) due to 1 previous error

Process finished with exit code 101

```

I was unable to figure out how to get it to work properly.